x/upgrade/keeper/keeper.go

```diff
@@ -49,6 +51,34 @@ func (k Keeper) SetUpgradeHandler(name string, upgradeHandler types.UpgradeHandl
 	k.upgradeHandlers[name] = upgradeHandler
 }
 
+// ProtocolVersionSetter defines the interface fulfilled by BaseApp
+// which allows setting it's appVersion field.
+type ProtocolVersionSetter interface {
+	SetProtocolVersion(uint64)
+}
```

Comment:
Should we put this into some `exported.go` file?

Reply:
yeah that sounds like a better place for it. [moved](https://github.com/cosmos/cosmos-sdk/pull/8897/files#diff-e420f8be7d4f239224d84df3f6a8956575f02c4d079a897b2a9bafe341dca690R1-R7)